tests/unittests/PartitionerTest.cpp

```diff
@@ -530,6 +533,87 @@ TEST_F(PartitionerTest, SimpleHeterogeneousPartitioning) {
   }
 }
 
+/// Test assigning more than one partitions in to one device.
+TEST_F(PartitionerTest, logicalIDTest0) {
+  auto *input1 =
+      mod_.createPlaceholder(ElemKind::FloatTy, {2, 10}, "input1", false);
+  auto *input2 =
+      mod_.createPlaceholder(ElemKind::FloatTy, {10, 16}, "input2", false);
+  auto *input3 =
+      mod_.createPlaceholder(ElemKind::FloatTy, {16, 20}, "input3", false);
+  auto *input4 =
+      mod_.createPlaceholder(ElemKind::FloatTy, {20, 1}, "input4", false);
+  auto *input5 =
+      mod_.createPlaceholder(ElemKind::FloatTy, {1, 50}, "input5", false);
+  auto *mul0 = F_->createMatMul("mul0", input1, input2);
+  auto *mul1 = F_->createMatMul("mul1", mul0, input3);
+  auto *mul2 = F_->createMatMul("mul2", mul1, input4);
+  auto *mul3 = F_->createMatMul("mul3", mul2, input5);
+  auto *save = F_->createSave("ret", mul3);
+  (void)save;
+  std::vector<DeviceInfo> devices = {{1500, BackendKind::Interpreter},
+                                     {1500, BackendKind::Interpreter}};
+  // Create two backends which support different ops, then do the partition by
+  // assigning the ops to the corresponding abackends.
+  auto partitioner = Partitioner(&mod_, devices, /* saturateHost */ true);
+  CompilationContext cctx;
+  auto err = partitioner.Partition(cctx);
+  EXPECT_FALSE(errToBool(std::move(err)));
+  DAGListTy myList = std::move(partitioner.getPartitionResult());
+  // Check there are 3 partitions.
+  ASSERT_EQ(mod_.getFunctions().size(), 3);
+  ASSERT_EQ(myList.size(), 1);
+  ASSERT_TRUE(checkSaveNode(mod_));
+
+  for (auto &dag : myList) {
+    // Check number of logical devices;
+    llvm::SmallSet<DeviceIDTy, 4> usedID;
+    for (auto &node : dag.nodes) {
+      ASSERT_EQ(node->logicalDevices.size(), 1);
+      usedID.insert(node->logicalDevices[0]);
+    }
+    // Check there are 2 logical devices.
+    ASSERT_EQ(usedID.size(), 2);
+  }
+  mod_.clear();
+}
+
+/// Test assigning more than one partitions in to one device.
```

Comment:
This test has the same comment as the one above, can you make the comments more specific. I'm a little confused what the difference is.

Reply:
Done. One test for single backendKind, one for heterogeneous partitioning.